sbc_compassion/models/correspondence_template.py

```diff
@@ -168,11 +172,72 @@ def _set_image(self):
                 })
                 self.template_attachment_id = attachment.id
 
-    @api.depends('template_image')
+    @api.onchange('template_image')
+    def _compute_size(self):
+        for template in self:
+            if template.template_image:
+                with tempfile.NamedTemporaryFile() as template_file:
+                    template_file.write(base64.b64decode(
+                        template.with_context(bin_size=False).template_image))
+                    template_file.flush()
+                    # Find the pattern inside the template image
+                    img = cv2.imread(template_file.name)
+                    template.page_height, template.page_width = img.shape[:2]
+                    template.qrcode_x_min = template.page_width * float(
+                        template.env['ir.config_parameter'].get_param(
+                            'qrcode_x_min'))
+                    template.qrcode_x_max = template.page_width * float(
```

Comment:
Maybe you can move this code inside `_compute_template_keypoints`method. and link it to the compute fields `qr_code`.
Then, inside this onchange method, you could simply call `self._compute_template_keypoints()`

It would be more clear no ? But you have to test it to see if it works.